modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotValidator.scala

```diff
@@ -141,6 +144,11 @@ object CurrencySnapshotValidator {
                   creationResult.focus(_.artifact.dataApplication).replace(expected.dataApplication)
               }
             }
+            .map { creationResult =>
+              if (creationResult.artifact.messages.isEmpty || creationResult.artifact.messages.exists(_.isEmpty))
```

Comment:
```suggestion
              if (creationResult.artifact.messages.forall(_.isEmpty))
```
scala> Option.empty[List[Int]].forall(_.isEmpty)
val res0: Boolean = true

scala> Option[List[Int]](List.empty).forall(_.isEmpty)
val res1: Boolean = true

scala> Option[List[Int]](List(1)).forall(_.isEmpty)
val res2: Boolean = false

Reply:
good tip, thanks :+1: